### ISSUE TITLE
plugin crashes Rack 0.6.2c

### ISSUE THREAD

**Issue by davephillips** [cclark2a/SchmickleWorks#12]
Fedora 23 Linux
VCV Rack 0.6.2c

Greetings ! I'm still building and trying to test your module but I have some persistent issues. The current build crashes Rack immediately, the plugin won't instantiate at all. And in previous builds when I did get the plugin loaded, I was never able to load a MIDI file. After dialing the GUI's pointer to Load it always simply sprang back without loading anything.

I'll post a more thorough bug report later today.

Best regards,

dp

**Comment by cclark2a** [comment]
I look forward to seeing more details. If you get any output from your terminal when it crashes, that would be very helpful.

About MIDI loading -- this is a difficult problem. I worked a little on fixing bugs with loading arbitrary MIDI files but ran into the problem that most files in the wild don't make sense for me. 

MIDI written for piano -- many notes on one channel that can sound simultaneously -- don't map well without substantial rewrites. MIDI written for single note instruments, or four part vocal arrangements, should be OK, but are harder to find.

That said, I have done little testing other than making sure that the MIDI files that NoteTaker writes can be read in again. I did just fix a bug where nearly all read files failed, which may be what you were running into on earlier builds.

I'll try running on Linux locally to see if I can reproduce the consistent failure. Thanks again!
